src/yunohost/tools.py

```diff
@@ -418,5 +413,90 @@ def tools_upgrade(auth, ignore_apps=False, ignore_packages=False):
         return { "log": service_log('yunohost-api', number="100").values()[0] }
+
+
+def tools_diagnosis(auth, private=False):
+    """
+    Return global info about current yunohost instance to help debugging
+
+    """
+    diagnosis = OrderedDict();
+
+    # Debian release
+    try:
+        with open('/etc/debian_version', 'r') as f:
+            debian_version = f.read().rstrip()
+    except IOError as e:
+        logger.warning(m18n.n('diagnosis_debian_version_error', error=format(e)), exc_info=1)
+    else:
+        diagnosis['host'] = "Debian %s" % debian_version
+
+    # Kernel version
+    try:
+        with open('/proc/sys/kernel/osrelease', 'r') as f:
+            kernel_version = f.read().rstrip()
+    except IOError as e:
+        logger.warning(m18n.n('diagnosis_kernel_version_error', error=format(e)), exc_info=1)
+    else:
+        diagnosis['kernel'] = kernel_version
+
+    # Packages version
+    diagnosis['packages'] = ynh_packages_version()
+
+    # Server basic monitoring
+    diagnosis['system'] = OrderedDict()
+    try:
+        disks = monitor_disk(units=['filesystem'], human_readable=True)
+    except MoulinetteError as e:
+        logger.warning(m18n.n('diagnosis_monitor_disk_error', error=format(e)), exc_info=1)
+    else:
+        diagnosis['system']['disks'] = {}
+        for disk in disks:
+            diagnosis['system']['disks'][disk] = 'Mounted on %s, %s (%s free)' % (
+                disks[disk]['mnt_point'],
+                disks[disk]['size'],
+                disks[disk]['avail']
+            )
+
+    try:
+        system = monitor_system(units=['cpu', 'memory'], human_readable=True)
+    except MoulinetteError as e:
+        logger.warning(m18n.n('diagnosis_monitor_system_error', error=format(e)), exc_info=1)
+    else:
+        diagnosis['system']['memory'] = {
+            'ram' : '%s (%s free)' % (system['memory']['ram']['total'], system['memory']['ram']['free']),
+            'swap' : '%s (%s free)' % (system['memory']['swap']['total'], system['memory']['swap']['free']),
+        }
+
+    # Services status
+    services = service_status()
+    diagnosis['services'] = {}
+    for service in services:
+        diagnosis['services'][service] = "%s (%s)" % (services[service]['status'], services[service]['loaded'])
+
+    # YNH Applications
+    try:
+        applications = app_list()['apps']
+    except MoulinetteError as e:
+        diagnosis['applications'] = m18n.n('diagnosis_no_apps')
+    else:
+        diagnosis['applications'] = {}
+        for application in applications:
+            if application['installed']:
+                diagnosis['applications'][application['id']] = application['label'] if application['label'] else application['name']
+
+    # Private data
+    if private:
+        diagnosis['private'] = OrderedDict()
+        # Public IP
+        diagnosis['private']['public_ip'] = {}
+        try: diagnosis['private']['public_ip']['IPv4'] = get_public_ip(4)
```

Comment:
split line here ("try:\n            diagnosis...") and same for following lines.